ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -863,8 +864,50 @@ let stream_of_channel : Lsp.Cli.Channel.t -> _ = function
     let sockaddr = Unix.ADDR_INET (Unix.inet_addr_loopback, port) in
     socket sockaddr
 
+(* Merlin uses [Sys.command] to run preprocessors and ppxes. We provide an
+   alternative version using the Spawn library for unixes.*)
+let run_in_directory ~prog ~prog_is_quoted:_ ~args ~cwd ?stdin:_ ?stdout
+    ?stderr:_ () =
+  (* Currently we assume that [prog] is always quoted and might contain
+     arguments such as [-as-ppx]. This is due to the way Merlin gets its
+     configuration. Thus we cannot rely on [Filename.quote_command]. *)
+  let args = String.concat ~sep:" " @@ List.map ~f:Filename.quote args in
+  let cmd = Format.sprintf "%s %s" prog args in
+
+  let prog = "/bin/sh" in
+  let argv = [ "sh"; "-c"; cmd ] in
+  let stdin = Unix.openfile "/dev/null" [ Unix.O_RDONLY ] 0x777 in
+  let stdout =
+    match stdout with
+    | Some file -> Unix.openfile file [ Unix.O_WRONLY ] 0x664
+    | None ->
+      (* Runned programs should never output to stdout since it is the channel
+         used by LSP to communicate with the editor *)
+      Unix.stderr
+  in
+  let pid =
+    let cwd : Spawn.Working_dir.t = Path cwd in
+    Spawn.spawn ~cwd ~prog ~argv ~stdin ~stdout ()
+  in
+  let _, status = Unix.waitpid [] pid in
+  let res =
+    match (status : Unix.process_status) with
+    | WEXITED n -> n
+    | WSIGNALED _ -> -1
+    | WSTOPPED _ -> -1
+  in
+  Unix.close stdin;
+  Unix.close stdout;
+  `Finished res
+
+let run_in_directory =
+  (* Merlin has specific stubs for Windows, we reuse them *)
+  let for_windows = !Merlin_utils.Std.System.run_in_directory in
```

Comment:
Can we stick to spawn on windows? it works quite well there.

Reply:
I think that currently it is not the case that ocaml-lsp uses spawn on windows: 

https://github.com/rgrinberg/merlin/blob/91fd966fdc00e259c2488127961daf3b7986a736/src/ocaml/driver/pparse.ml#L44-L65

I simply replicated (hopefully) the current behaviour.